src/sentry/integrations/github_enterprise/repository.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from sentry.models import Integration
+from sentry.plugins import providers
+
+
+WEBHOOK_EVENTS = ['push', 'pull_request']
+
+
+class GitHubEnterpriseRepositoryProvider(providers.IntegrationRepositoryProvider):
```

Comment:
it also seems like a lot of this could be shared with the github repo provider. that doesn't have to be a blocker for getting this in, but if you can comment/leave notes as to where they differ, that'd probably be helpful